python/ray/air/integrations/mlflow.py

```diff
@@ -1,15 +1,179 @@
 import logging
-from typing import Dict, Optional
+from types import ModuleType
+from typing import Dict, Optional, Union
 
 import ray
+from ray.air import session
+
 from ray.air._internal.mlflow import _MLflowLoggerUtil
 from ray.tune.logger import LoggerCallback
 from ray.tune.result import TIMESTEPS_TOTAL, TRAINING_ITERATION
 from ray.tune.experiment import Trial
+from ray.util.annotations import PublicAPI
+
+try:
+    import mlflow
+except ImportError:
+    mlflow = None
+
 
 logger = logging.getLogger(__name__)
 
 
+class _NoopModule:
+    def __getattr__(self, item):
+        return _NoopModule()
+
+    def __call__(self, *args, **kwargs):
+        return None
+
+
+@PublicAPI(stability="alpha")
+def setup_mlflow(
+    config: Optional[Dict] = None, rank_zero_only: bool = True, **kwargs
```

Comment:
why does config need to be passed in here if all the configuration is passed in through kwargs? Is this primarily for backwards compatibility?

Reply:
I would prefer to actually have all the configuration args listed explicitly in the API instead of kwargs, and don't require that configurations need to be passed in via the `config` dict.

Reply:
I had this in a previous iteration and the reason I went to `config` was mainly to keep consistency with `wandb_setup`. 

I'm happy to move back to pure kwargs and not parsing a config, but I think we should then also do this for weights and biases

Reply:
I guess the main reason we do this for W&B is because we pass the other config parameters as run metadata (`wand.init` takes a `config` parameter). We could do the same in mlflow to justify the config

Reply:
I've updated the PR to also log parameters. Let's have a chat offline which API we want to go with

Reply:
Context:

- We’ve got rid of the @wandb_mixin decorator in favor of a setup_wandb(config) function for W&B
- We’re doing the same for MlFlow
- The question is, should we use a config-API setup_mlflow(config) or  and explicit API setup_mlflow(experiment_name, experiment_id, tracking_uri, …) 

Pro config:
- Same way we used to configure the mixins
- Required for W&B so it logs the config to the run (otherwise users have to do this manually. They are used to passing it to wandb.init() )
- We should be consistent between wandb and mlflow mixins

Pro args:
- Explicit about which args are supported
- Passing the mixin configs via config was a workaround because this was the easiest way to pass these settings and still retain some flexibility. But with the explicit setup_xyz() APIs this workaround isn’t needed anymore

Proposal:
- How about we go with setup_mlflow(config, experiment_name, experiment_id) (similar for wandb). If a mlflow or - wandb config key is set, we print a deprecation warnings and tell people to pass the configs as proper arguments instead.
- This will get rid of the “special” config keys but we can still log the rest of the config automatically.